met-web/src/components/engagement/form/EngagementWidgets/Documents/FileDrawer.tsx

```diff
@@ -42,25 +45,63 @@ const FileDrawer = () => {
 
     const { handleSubmit, reset } = methods;
 
+    useEffect(() => {
+        methods.setValue('name', documentToEdit?.title || '');
+        methods.setValue('link', documentToEdit?.url || '');
+        methods.setValue('folderId', documentToEdit?.parent_document_id || 0);
+    }, [documentToEdit]);
+
     const handleClose = () => {
         reset();
         handleFileDrawerOpen(false);
     };
 
+    const updateDocument = async (data: FileForm) => {
+        if (!(documentToEdit && widget)) {
+            return;
+        }
+        const documentEditsToPatch = updatedDiff(documentToEdit, {
+            title: data.name,
+            parent_document_id: data.folderId === 0 ? null : data.folderId,
+            url: data.link,
+        }) as PatchDocumentRequest;
+        await patchDocument(widget.id, documentToEdit.id, {
+            ...documentEditsToPatch,
+            parent_document_id: data.folderId === 0 ? null : data.folderId,
```

Comment:
but this way we are always sending parent_document_id, can we have it that we only send it if it changed

Reply:
I swear I changed these on my local cleaning that up now